app/angular/src/client/preview/angular/components/app.component.ts

```diff
@@ -51,7 +53,7 @@ export class AppComponent implements OnInit, OnDestroy {
       );
 
       this.subscription = this.data.subscribe((newData) => {
-        this.setProps(instance, newData);
+        this.ngZone.run(() => this.setProps(instance, newData));
```

Comment:
Is it really necessary to call `ngZone.run`? I thought the two following lines already take care that the updated props are properly updated by the change detection 

```typescript
        childChangeDetectorRef.markForCheck();
        // Must detect changes on the current component in order to update any changes in child component's @HostBinding properties (angular/angular#22560)
        this.changeDetectorRef.detectChanges();
```

Though, I must admit I rarely need to leave or re-enter ngZone.

Do you have an example of what does not work right now and would work after this change?

Reply:
That is correct that updates made to the props during that tick would get detected, but this is about functions initialized during the tick.

I am not an expert on how the zones work, so I may not be explaining it exactly right. For change detection to know when to trigger a detection tick, it hooks functions like `addEventListener`, `setTimeout`, Promises and I'm sure some others. They only get hooked while inside a zone, so when a knob updates a prop the channel triggers the event from a non-hooked function and we update the props without being in the zone. Anything changed from setting the props would be detected, since change detection was manually triggered, but if we subscribe to an output then that event will not be hooked, so when the output EventEmitter emits, Angular don't know about it and will not trigger a change detection tick.

I do think I may be doing to much in the zone by calling it there, instead of maybe the parts inside `setProps` that interact with the component, but I will try to come up with example that is more clear, but the one that caused me to recognize the problem is in [this](https://github.com/storybookjs/storybook/issues/7242#issuecomment-613008630) comment. The click event is hooked, since the initial setup is done in the zone, to clicking the button is detected. If the `visible` knob is clicked, the button is removed and if the knob is clicked again it will correctly show the button again, since change detection was manually triggered after setting the props. If you click the button in the story, the click event no longer trigger's a change detection tick, since the subscription was done outside the zone.